src/libraries/System.Private.Uri/src/System/Uri.cs

```diff
@@ -2377,11 +2378,12 @@ private unsafe void CreateUriInfo(Flags cF)
         Done:
             cF |= Flags.MinimalUriInfoSet;
             info.DnsSafeHost = _dnsSafeHost;
-            lock (_string)
+
+            Interlocked.CompareExchange(ref _info, info, null!);
```

Comment:
We previously only assigned _info if MinimalUriInfoSet wasn't specified.  Why is it ok to now always assign it?

Reply:
`_info` and `MinimalUriInfoSet` are only set in `CreateUriInfo`.
`MinimalUriInfoSet` being set == `_info` being set. I suppose the flag is not really necesarry as we can do a null check on `_info` instead.
The Interlocked operation guarantees that `_info` is only being written to once.

For a brief moment it could happen that `_info` is set but the flag is not yet set, but that doesn't matter as we always access `_info` after a call to `EnsureUriInfo` that checks the flag.

Reply:
> MinimalUriInfoSet being set == _info being set.

Then why do we need the flag?

Reply:
> Then why do we need the flag?

If we have a local copy of Flags and check that MinimalUriInfoSet is set, we can be sure that `_info` and some of its fields are set **and** flags from `CreateUriInfo` are set.

If we read `_info` and see it's not null, we can be sure its fields are set, **but not** that all the relavant flags have been set.

The alternative to this would be to set the flags and _info *at the same time* under a lock, but we have nothing useful to lock to yet at this point.